system/gatekeeper/templates/constrainttemplate-resource-limits.yaml

```diff
@@ -13,54 +13,34 @@ spec:
       libs:
         - |
           {{ .Files.Get "lib/add-support-labels.rego" | nindent 10 }}
+        - |
+          {{ .Files.Get "lib/traversal.rego" | nindent 10 }}
       rego: |
         package resourcelimits
         import data.lib.add_support_labels
+        import data.lib.traversal
 
         iro := input.review.object
+        pod := traversal.find_pod(iro)
 
         # We have such a bafflingly large amount of violations that
         # Gatekeeper cannot report them all at once. I'd rather have
         # *something* reported for as many things as possible, even if it
         # means sacrificing accuracy, so we report only one violation for
-        # each offending pod, and we report on the deployment/daemonset level
-        # if possible to avoid useless duplication of violations.
+        # each offending pod or pod owner.
 
         resource_types = {"cpu", "memory"}
 
-        missing_limits_on_pod[{"container": container.name, "type": resource_type}] {
-          container := iro.spec.containers[_]
+        missing_limits[{"container": container.name, "type": resource_type}] {
+          pod.found
```

Comment:
I am not super happy about this name. Maybe we find something more descriptive like ``pod.isFound`` or ``pod.found?``.